components/x-privacy-manager/src/privacy-manager.jsx

```diff
@@ -0,0 +1,119 @@
+/**
+ * @typedef {{ ok: boolean, status?: number }} _Response
+ */
+
+import { h } from '@financial-times/x-engine';
+import { withActions } from '@financial-times/x-interaction';
+
+import s from './privacy-manager.scss';
+import { RadioBtn } from './radio-btn';
+import { LoadingMessage, ResponseMessage } from './messages';
+
+// TODO: Replace with the genuine endpoint
+export const CONSENT_API = 'https://consent.ft.com';
+
+export const withCustomActions = withActions(() => ({
+	onConsentChange() {
+		return ({ consent = true }) => ({ consent: !consent });
+	},
+
+	sendConsent() {
+		return async ({ isLoading, consent }) => {
+			if (isLoading) return;
+
+			const body = JSON.stringify({
+				demographic: consent,
+				behavioural: consent,
+				programmatic: consent
+			});
+
+			try {
+				const res = await fetch(CONSENT_API, { method: 'PATCH', body });
+				return { _response: { ok: res.ok } };
+			} catch (err) {
+				return { _response: { ok: false } };
+			}
+		};
+	}
+}));
+
+/**
+ * @param {boolean} isLoading
+ * @param {_Response} response
+ * @param {string} referrer
+ */
+function renderMessage(isLoading, response, referrer) {
+	if (isLoading) return <LoadingMessage />;
+	if (response) return <ResponseMessage success={response.ok} referrer={referrer} />;
+	return null;
+}
+
+/**
+ * @param {{
+ *   consent?: boolean
+ *   referrer?: string
+ *   legislation?: string[]
+ *   actions: {
+ *     onConsentChange: () => void
+ * 	   sendConsent: () => Promise<{_response: _Response }>
+ *   },
+ *   isLoading: boolean
+ *   _response?: _Response
+ * }} Props
+ */
+export function BasePrivacyManager({
+	consent = true,
+	referrer,
+	actions,
+	isLoading,
+	_response = undefined
+}) {
+	return (
+		<div className={s.consent}>
+			<h1 className={s.consent__title}>Do Not Sell My Personal Information</h1>
+			<div className={s.consent__copy}>
+				<p>
+					CCPA defines the "sale" of personal information in extremely broad terms. For this reason,
+					the collection of data by advertisers who advertise on our Sites for the purposes of
+					measuring the effectiveness of their advertising is caught, and requires the ability to
+					object to the use of your personal information. The data collected by advertisers may
+					include online identifiers, such as IP address, and interactions with advertisements.
+				</p>
+				<p>
+					Our advertising Terms and Conditions limit the use of this data by advertisers only to
+					their own business purposes, including analytics and attribution, not for commercial
+					purposes.
+				</p>
```

Comment:
I think I have seen the `next-fow-api` mentioned in relation to this work. Is it a TODO to get this text from there?

Reply:
Yes, but we may leave this hardcoded this for the time being if it helps us meet our July 1st deadline